src/rust/engine/tar_api/src/tar_api.rs

```diff
@@ -0,0 +1,15 @@
+extern crate flate2;
+extern crate tar;
+
+use flate2::read::GzDecoder;
+use std::fs::File;
+use std::path::Path;
+use tar::Archive;
+
+pub fn decompress_tgz(tar_path: &Path, output_dir: &Path) -> Result<(), std::io::Error> {
```

Comment:
Any chance of a quick unit test? Just checking in a `.tar.gz` and showing that it unpacks properly...

Reply:
Test added.